src/app/core/services/auth/auth.service.ts

```diff
@@ -0,0 +1,51 @@
+import { Injectable } from "@angular/core";
+import { HttpClient } from "@angular/common/http";
+import { BehaviorSubject, Observable } from "rxjs";
+import { map } from "rxjs/operators";
+import { IUser } from "./../../../shared/interfaces/user";
+import { apiUrl } from "./../../../../environments/environment";
+
+@Injectable({
+  providedIn: "root"
+})
+export class AuthService {
+  private currentUserSubject: BehaviorSubject<IUser>;
+  public currentUser: Observable<IUser>;
+
+  constructor(private http: HttpClient) {
```

Comment:
This login should be mover into auth guard and you can also use 'jwt helper' to check no only existing of jwt toket, but also is expired or not